Fails to build with Clang/LLVM 22

https://github.com/llvm/llvm-project/issues/167709#issuecomment-4021564275

```
In file included from ../src/dxvk/dxvk_pipemanager.cpp:1:
In file included from /home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/optional:269:
In file included from /home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/__format/range_format.h:19:
In file included from /home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/__format/fmt_pair_like.h:14:
In file included from /home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/__fwd/pair.h:14:
/home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/__fwd/tuple.h:40:52: error: a parameter pack may not be accessed at an out of bounds index
   40 |   using type _LIBCPP_NODEBUG = __type_pack_element<_Ip, _Tp...>;
```

```
In file included from ../src/dxvk/dxvk_pipemanager.cpp:1:
In file included from /home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/optional:1805:
In file included from /home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/iterator:755:
In file included from /home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/variant:1647:
/home/fedora/llvm-mingw-polly/x86_64-w64-mingw32/include/c++/v1/tuple:1120:52: error: no member named '__base_' in 'std::tuple<>'
 1120 |   return static_cast<__tuple_leaf<_Ip, type>&>(__t.__base_).get();
```

This is supposedly fixed in the `release/22.x` branch however I am still getting this issue. I am not sure if this is strictly an LLVM/Clang bug still or an issue with DXVK so I am posting this here as well.

Thanks!

--- a/src/dxvk/dxvk_pipemanager.cpp
+++ b/src/dxvk/dxvk_pipemanager.cpp
@@ -380,7 +380,7 @@ namespace dxvk {
 
     auto iter = m_shaderLibraries.emplace(
       std::piecewise_construct,
-      std::tuple(),
+      std::tuple(key),
       std::tuple(m_device, this, key));
     return &iter.first->second;
   }

